components/motor/gpio/encoded.go

```diff
@@ -159,14 +158,13 @@ func (m *EncodedMotor) makeAdjustments(ctx context.Context, goalRPM, goalPos, di
 			continue
 		}
 		now := time.Now().UnixNano()
-
-		if (direction == 1 && pos >= goalPos) || (direction == -1 && pos <= goalPos) {
+		if (goalPos-currentTicks)*direction < 0 {
 			// stop motor when at or past goal position
 			return m.Stop(ctx, nil)
 		}
 
 		// calculate RPM based on change in position and change in time
-		deltaPos := (pos - lastPos) / m.ticksPerRotation
+		deltaPos := (currentTicks - lastTicks) / m.ticksPerRotation
```

Comment:
... can we actually just calculate speed in ticks per second, or is it too far gone from physical intuition?

Reply:
i liked leaving this math as rpm here(because real world), but we could just as easily move the math to `encodedGoForMath` and use ticks per second everywhere. having physical intuition is nice tho

Reply:
Let's keep physical intuition.